railgun/internal/ctrlutils/utils.go

```diff
@@ -24,7 +24,7 @@ const KongIngressFinalizer = "configuration.konghq.com/ingress"
 // UpdateKongAdmin is a helper function to take the contents of a Kong config and update the Admin API with the parsed contents.
 func UpdateKongAdmin(ctx context.Context, kongCFG *sendconfig.Kong) error {
 	// build the kongstate object from the Kubernetes objects in the storer
-	storer := store.New(*mgrutils.CacheStores, "kong", false, false, false, logrus.StandardLogger())
+	storer := store.New(*mgrutils.CacheStores, mgrutils.IngressClass, false, false, false, logrus.StandardLogger())
```

Comment:
`false, false, false` here means that KIC 2.0 will drop support for the legacy IngressClass mode of operation.

@hbagdi do you think it's a reasonable time to drop the knob allowing classless resources?
We [said](https://github.com/Kong/kubernetes-ingress-controller/blob/main/CHANGELOG.md#0100---20200915) we require setting the ingress class field by default in 0.10 (7 months ago).

If yes: the action item is to start drafting a changelog for 2.0 and mentioning that we drop the `process-classless-*` flags and the respective functionality.

If not: the action item is to add these `process-classless-*` flags to Railgun in this PR.

Reply:
I'm in favor of dropping the support and adding that to the release notes.

Reply:
Is there a compelling reason to drop it? Does continued maintenance impose a burden beyond addition of the flags?

If we can continue to use the existing code for the legacy classless logic, not dropping it provides a simpler upgrade path to 2.0 if you've never bothered to add classes (you should, but alas).

The natural breaking point seems like adding logic to handle the [v1 default IngressClass annotation](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class) (and broader support for IngressClass objects in general), since that's at least incompatible with the v1 flag. v1beta1 doesn't have any such equivalent, though it's going away entirely. KongConsumer we're free to do whatever.

Reply:
For now I've turned all of these back on, let me know if that satisfies for the purposes of this PR, and perhaps we need a follow up to take another look at this in the future? :thinking:

Reply:
These should be turned off by default - because they have been since KIC 0.10.

What we should be deciding upon is whether we're dropping support for flipping individual `bool`s to `true` using the below flags:

https://github.com/Kong/kubernetes-ingress-controller/blob/fa0afebe1ccd19551305c89008d829a10790a1a0/cli/ingress-controller/flags.go#L166-L171

Reply:
My bad, I had a misunderstanding. The latest commits reflect this: flags enabled, defaults set and its fed through the controllers to the `KongAdminUpdate()` :+1:

Reply:
I would like to keep us moving and unblocked so now that the behavior is identical to 1.x I'm going to consider this one resolved, please unresolve if there's any further issues or follow up you feel needs to be addressed. :bow: